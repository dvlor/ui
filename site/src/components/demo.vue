<script>
import { defineComponent, useSlots } from 'vue'

export default defineComponent({
  setup(props, { slots }) {
    return {
      leftChild: slots.default().filter((c, i) => i % 2 == 0),
      rightChild: slots.default().filter((c, i) => i % 2)
    }
  }
})
</script>
<template>
  <div class="demo">
    <div class="left">
      <component class="demo-item" v-for="c in leftChild" :is="c" />
    </div>
    <div class="right">
      <component class="demo-item" v-for="c in rightChild" :is="c" />
    </div>
  </div>
</template>
<style lang="less">
.demo {
  display: flex;
  .left {
    flex: 1;
    margin-right: 15px;
    overflow: auto;
  }
  .right {
    flex: 1;
    overflow: auto;
  }
  .demo-item {
    margin-bottom: 15px;
  }
}
</style>
