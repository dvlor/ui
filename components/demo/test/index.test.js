import { Demo } from '../demo'
import { render, fireEvent } from '@testing-library/vue'

describe('components/demo', () => {
  it('render', () => {
    expect(render(<Demo>click</Demo>).html()).toMatchSnapshot()
  })

  it('click event', async () => {
    let i = 0
    const handleClick = () => {
      i++
    }

    const { getByText } = render(<Demo onClick={handleClick}>click me</Demo>)
    await fireEvent.click(getByText('click me'))
    expect(i).toBe(1)
  })
})
