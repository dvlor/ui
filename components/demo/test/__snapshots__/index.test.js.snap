// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/demo render 1`] = `"<div class=\\"u--demo\\">click</div>"`;
